packages/server/lib/hooks/hooks.ts

```diff
@@ -185,7 +187,9 @@ export const connectionRefreshFailed = async ({
 
     const slackNotificationService = new SlackService({ orchestratorClient: getOrchestratorClient(), logContextGetter });
 
-    void slackNotificationService.reportFailure(connection, connection.connection_id, 'auth', logCtx.id, environment.id, config.provider);
+    const randomDelay = Math.random() * 1000;
+    await delay(randomDelay);
```

Comment:
seems unnecessary if you wrap the whole hook inside a transaction

Reply:
Tested using wrapping the hook inside a transaction and it doesn't fix the multiple slack notification issue